### ISSUE TITLE
Room/Bed dropdown on patient details view

### ISSUE THREAD

**Issue by MaxSchaefer** [helpwave/web#473]
Replace the current bed index indicator with the new two Room/Bed dropdowns. 

> Adding two additional drop down controls to assign a room and a bed
The room drop down needs the option "unassigned" when that is selected the bed drop down is inactive

<img width="339" alt="250274074-5c3b7f9c-095b-473e-8433-777bb00b971e" src="https://github.com/helpwave/web/assets/40192999/30c7472c-1f6b-4b36-a8e9-765b7eb9ddb7">

https://github.com/helpwave/design/issues/2

**Comment by DasProffi** [comment]
Blocked by #486, because the functionality of this page is overhauled in that PR

**Comment by DasProffi** [comment]
What is the intended behavior of this?
Is the component only showing free beds, or are also taken beds allowed, here it would be useful to have an optimized request only showing:
```
rooms: {
	id: string,
	name: string,
	beds: {
		id: string,
		index: number,
		patientID?: string
	}
}
```

In any case, it also would be nice to have a reassign patient method, so we do not have to make two request for unassign and reassign

**Comment by MaxSchaefer** [comment]
The method should return your proposed structure by a ward id, correct? A similar method is `GetRoomsByWard` but this method does not return a patient.

> In any case, it also would be nice to have a reassign patient method, so we do not have to make two request for unassign and reassign
You could just call `AssignBed`. It will override a previous assignment.

**Comment by DasProffi** [comment]
> The method should return your proposed structure by a ward id, correct? A similar method is `GetRoomsByWard` but this method does not return a patient.

But this does not include the patient as you mentioned and without that information it isn't possible to make a good design for the dropdown
> You could just call `AssignBed`. It will override a previous assignment.

But is that the behavior we want? It seems very easy to accidentally unassign a patient

In my opinion an indication whether the bed is taken is necessary and therefore such an endpoint as described above

**Comment by MaxSchaefer** [comment]
> But is that the behavior we want? It seems very easy to accidentally unassign a patient

We could change the behavior of the `AssignBed` method. Right now, this is the behavior.

> In my opinion an indication whether the bed is taken is necessary and therefore such an endpoint as described above

100%, sure! Therefore, I've opened an issue https://github.com/helpwave/services/issues/269

**Comment by DasProffi** [comment]
How does it work when change the room? The bed has to change, but to what? undefined/empty would be a possibility, but then we need an indication when a bed swap is applied and when not and should it also be possible to unassign a patient this way?

**Comment by DasProffi** [comment]
Solution is to use a saved/unsaved tag below the dropdowns and a revert button to the right to indicate the reversal of changes
